pkg/container-collection/k8s.go

```diff
@@ -181,3 +192,65 @@ func (k *K8sClient) ListContainers() (arr []Container, err error) {
 	}
 	return arr, nil
 }
+
+func getSocketPathFromKubelet(clientset *kubernetes.Clientset, nodeName string) (string, error) {
+	// In case there's an environment variable, it overrides the detection
+	envsps := []string{"INSPEKTOR_GADGET_DOCKER_SOCKETPATH", "INSPEKTOR_GADGET_CONTAINERD_SOCKETPATH", "INSPEKTOR_GADGET_CRIO_SOCKETPATH", "INSPEKTOR_GADGET_PODMAN_SOCKETPATH"}
+
+	for _, envsp := range envsps {
+		if os.Getenv(envsp) != "" {
+			return "", nil
+		}
+	}
+	kubeletConfig, err := getCurrentKubeletConfig(clientset, nodeName)
+	if err != nil {
+		return "", fmt.Errorf("getting /configz: %w", err)
+	}
+	socketPath, found := strings.CutPrefix(kubeletConfig.ContainerRuntimeEndpoint, "unix:///")
```

Comment:
I still think  `///` is wrong: 

```
# notice missing leading / for the path:
time="2024-05-28T16:50:30Z" level=info msg="using the detected container runtime socket path from Kubelet's config: var/run/cri-dockerd.sock"
```

Probably it works later on because a path.Join operation adds the missing `/`.